### DISCUSSION TITLE
Problem with gateways and microservices deployment in IIS

### DISCUSSION THREAD

**Discussion by fedepaka** [Burgyn/MMLib.SwaggerForOcelot#233] [Q&A]
Error summary: Failed to load API definition. Fetch error
response status is 404 /swagger/docs/1.0/apiweb,  in Api Gateway.

I have a microservice published in IIS. I share a screenshot of it running: 

![image](https://user-images.githubusercontent.com/36859945/178570544-677a31ed-3010-4232-b650-864915ccda92.png)

Then I do in deploy in IIS of the ApiGateway, but when I want to get the definition of the endpoints of the microservice, I get the error as in the screenshot:

![image](https://user-images.githubusercontent.com/36859945/178572379-dbbd27bf-60fb-49e5-a1dc-27ff802c3b96.png)

 Next, Ocelot.json configuration file
```json
 {
 
  "Routes": [

    {
      "DownstreamPathTemplate": "/WebApiCatalogue.ApiService/api/catalogue", 
      "DownstreamScheme": "https",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 8091
        }
      ],
      "UpstreamPathTemplate": "/api/catalogue", 
      "UpstreamHttpMethod": ["GET"],
      "SwaggerKey": "apiweb"
    }
  ],
  "SwaggerEndPoints": [
    {
      "Key": "apiweb",
      "TransformByOcelotConfig": true,
      "Config": [
        {
          "Name": "API WEb WebApiCatalogue.ApiService",
          "Version": "1.0",
          "Url": "https://localhost:8091/WebApiCatalogue.ApiService/swagger/v1/swagger.json"
        }
      ]
    }
  ],
  "GlobalConfiguration": {
    "BaseUrl": "https://localhost:7014"

  }
}
```

The configuration method

```c#
app.UseSwaggerForOcelotUI(opt =>
{
     opt.PathToSwaggerGenerator = "/swagger/docs";
});

app.UseRouting();

app.UseEndpoints(async endpoints =>
{
     endpoints.MapGet("/", async context =>
     {
            await context.Response.WriteAsync("The Api Gateway is working");
     });
});
app.UseOcelot().Wait();
```

However, if I run the APIGateway from Visual Studio, it runs without problems and shows the Microservice definition. Attached is a screenshot.

![image](https://user-images.githubusercontent.com/36859945/178575387-58f4d27b-7a9d-43cd-9746-e49c21feeb27.png)

I would be grateful if you could help me with this.
Best regards

**Comment by Burgyn** [comment]
Hi.  I'm currently away from PC for a few days and can't look into it.  When you run on IIS, do you also use port 8091 for the downstream service?

**Reply by fedepaka** [reply]
Hi Burgyn, thank you very much for your reply.
The port used for API Gateway access is 443, and the port where the Micro Service is running is 8091.
Thank you very much for your help

**Reply by fedepaka** [reply]
Hi Burgyn, how are you?
Any comments on my previous question?

**Reply by Burgyn** [reply]
Hi @fedepaka,
thanks I'm fine and you?

At first glance, your configuration looks good and I don't see any major problem there.

Do you have any logs from the IIS server? It would be ideal if you could prepare a demo project where you would isolate this problem. I can look into it then.
